pkg/clusterconditions/clusterconditions.go

```diff
@@ -104,3 +108,32 @@ func (r *conditionRegistry) Match(ctx context.Context, matchingRules []configv1.
 
 	return false, errors.NewAggregate(errs)
 }
+
+type PromQLTarget struct {
+	KubeClient      kubernetes.Interface
+	UseDNS          bool
+	URL             *url.URL
+	CABundleFile    string
+	BearerTokenFile string
+
+	// Kubernetes service that contains the name and the namespace to be used to resolve
+	// the IP address when DNS is disabled.
+	KubeSvc *corev1.Service
+}
+
+func DefaultPromQLTarget() PromQLTarget {
+	promqlURL, _ := url.Parse("https://thanos-querier.openshift-monitoring.svc.cluster.local:9091")
+	return PromQLTarget{
+		KubeClient: nil,
+		UseDNS:     false,
+		URL:        promqlURL,
+		KubeSvc: &corev1.Service{
```

Comment:
I don't think we need a full `corev1.Service`, all we need is namespace/name. Apimachinery has a `NamespacedName` type for this:

https://github.com/openshift/cluster-version-operator/blob/master/vendor/k8s.io/apimachinery/pkg/types/namespacedname.go#L27-L30

Reply:
Ah, yes! That's perfect for this use case, thanks!

Replaced in the new commit.